src/wrappers/themis/wasm/test/test.js

```diff
@@ -77,4 +77,214 @@ describe('wasm-themis', function() {
             })
         })
     })
+    describe('SecureCell', function() {
+        let masterKey1 = new Uint8Array([1, 2, 3, 4])
+        let masterKey2 = new Uint8Array([5, 6, 7, 8, 9])
+        let emptyArray = new Uint8Array()
+        let testInput = new Uint8Array([1, 1, 2, 3, 5, 8, 13])
+        let testContext = new Uint8Array([42])
+        describe('Seal mode', function() {
+            it('does not accept strings', function() {
+                // It's a really nice idea to accept strings as 'master keys', but we have
+                // to define their interpretetaion. So treat them as errors for now.
+                assert.throws(() => new themis.SecureCellSeal('master key'), TypeError)
+            })
+            it('encrypts without context', function() {
+                let cell = new themis.SecureCellSeal(masterKey1)
+                let encrypted = cell.encrypt(testInput)
+                let decrypted = cell.decrypt(encrypted)
+                assert.deepStrictEqual(decrypted, testInput)
+            })
+            it('encrypts with context', function() {
+                let cell = new themis.SecureCellSeal(masterKey1)
+                let encrypted = cell.encrypt(testInput, testContext)
+                let decrypted = cell.decrypt(encrypted, testContext)
+                assert.deepStrictEqual(decrypted, testInput)
+            })
+            it('produces extended results', function() {
+                let cell = new themis.SecureCellSeal(masterKey1)
+                let encrypted = cell.encrypt(testInput)
+                assert(encrypted.length > testInput.length)
+            })
+            it('forbits empty inputs', function() {
+                assert.throws(() => new themis.SecureCellSeal(emptyArray),
+                    expectError(ThemisErrorCode.INVALID_PARAMETER)
+                )
+                let cell = new themis.SecureCellSeal(masterKey1)
+                assert.throws(() => cell.encrypt(emptyArray),
+                    expectError(ThemisErrorCode.INVALID_PARAMETER)
+                )
+            })
+            it('empty context == no context', function() {
```

Comment:
check for null?

Reply:
Currently we do not accept explicit `null` or `undefined`, they are treated as type errors.

- ok: `cell.encrypt(msg)`
- ok: `cell.encrypt(msg, context)`
- ok: `cell.encrypt(msg, new Uint8Array())` — equivalent to `cell.encrypt(msg)`
- error: `cell.encrypt(msg, null)`
- error: `cell.encrypt(msg, undefined)`

What would be the desired behavior? I believe that `undefined` should still be an error, but we may start treating `null` similar to an empty array or an omitted argument.

Reply:
yes, it's desired behaviour, i suggest to add checks for `null` and `undefined` to test suit

Reply:
I've allowed `null` context values for Sealing and Token Protect modes, they are treated as empty context now. Context Imprint mode will not accept null (and empty context too).

undefined is treated as type error. The tests for this will be coming in a separate PR.